test/e2e/specs/editor/various/post-title.spec.js

```diff
@@ -0,0 +1,363 @@
+				`,
+			} );
+			await pageUtils.pressKeys( 'primary+v' );
+
+			// Check the HTML elements have been stripped from the first block's
+			// `content` attribute...
+			await expect( pageTitleField ).toHaveText(
+				'I am heading block title with HTML tag'
+			);
+
+			// ...and are not rendered.
+			await expect( pageTitleField.locator( 'css=strong' ) ).toBeHidden();
+
+			// Check the 2nd block ended up in the post contents and did not
+			// have its HTML stripped out.
+			await expect.poll( editor.getBlocks ).toMatchObject( [
+				{
+					name: 'core/paragraph',
+					attributes: {
+						content:
+							'And I am the rest of titles with <em>emphasis tag</em>!',
+					},
+				},
+			] );
+		} );
+
+		test.skip( 'should output HTML tags in plaintext when added into Post Title field in visual editor mode', async ( {
+			editor,
+			page,
+			admin,
+			pageUtils,
+		} ) => {
+			await admin.createNewPost();
+
+			const pageTitleField = editor.canvas.getByRole( 'textbox', {
+				name: 'Add title',
+			} );
+
+			await expect( pageTitleField ).toBeFocused();
+
+			await page.keyboard.type( 'I am <em>emphasis</em>' );
+
+			await expect( pageTitleField ).toHaveText(
+				'I am <em>emphasis</em>'
+			);
+
+			// Check that the `em` tag was output in plaintext and not rendered.
+			await expect( pageTitleField.locator( 'css=em' ) ).toBeHidden();
+
+			// Switch to code view
+			await pageUtils.pressKeys( 'secondary+M' ); // Emulates CTRL+Shift+Alt + M => toggle code editor
+
+			const codeViewPageTitleField = editor.canvas.getByRole( 'textbox', {
+				name: 'Add title',
+			} );
+
+			// Check that the `em` tag was output in plaintext (HTML entities)
+			await expect( codeViewPageTitleField ).toHaveText(
+				'I am &lt;em&gt;emphasis&lt;/em&gt;'
+			);
+		} );
+
+		test( 'should output HTML tags in plaintext in visual editor mode when HTML is added in plaintext in code editor mode', async ( {
+			editor,
+			page,
+			admin,
+			pageUtils,
+		} ) => {
+			await admin.createNewPost();
+
+			// switch Editor to code editor mode
+			// Open code editor
+			await pageUtils.pressKeys( 'secondary+M' ); // Emulates CTRL+Shift+Alt + M => toggle code editor
+
+			// Check we're in Code view mode.
+			await expect(
+				editor.canvas.getByRole( 'heading', {
+					name: 'Editing code',
+				} )
+			).toBeVisible();
+
+			const codeViewPageTitleField = editor.canvas.getByRole( 'textbox', {
+				name: 'Add title',
+			} );
+
+			await codeViewPageTitleField.focus();
+
+			// Also verifies that the field handles typing into the field.
+			await page.keyboard.type( 'I am &lt;em&gt;emphasis&lt;/em&gt;' );
+
+			await expect( codeViewPageTitleField ).toHaveText(
+				'I am &lt;em&gt;emphasis&lt;/em&gt;'
+			);
+
+			// Switch to visual view
+			await pageUtils.pressKeys( 'secondary+M' ); // Emulates CTRL+Shift+Alt + M => toggle code editor
+
+			// Wait for canvas iframe to be ready otherwise queries below will fail.
```

Comment:
@WunderBart I wonder if this will no longer be necessary after you merged https://github.com/WordPress/gutenberg/pull/54911 ?

Reply:
Looking at the test, that part shouldn't be necessary at all since we're already interacting with the canvas above. So, if `canvas` is present and the `getBy*` locator is lazily evaluated (auto-waiting for the content), we should be safe without the part in question.

In case it's still failing though, what is the exact error?

Reply:
It cannot find the element at all. 

```
Error: Timed out 5000ms waiting for expect(received).toHaveText(expected)

    Expected string: "I am <em>emphasis</em>"
    Received string: ""
    Call log:
      - expect.toHaveText with timeout 5000ms
      - waiting for getByRole('textbox', { name: 'Add title' })
      - waiting for getByRole('textbox', { name: 'Add title' })


      355 |
      356 |                     // Check that the `em` tag was output in plaintext
    > 357 |                     await expect( visualViewPageTitleField ).toHaveText(
          |                                                              ^
      358 |                             'I am <em>emphasis</em>'
      359 |                     );
```

Reply:
Note that we are switching from "Code Editor" mode to "Visual Editor" mode. I doubt that flow is heavily tested and so this may be a new bug.

Reply:
I'd imagine it's because in whatever test setup provides us with `editor.canvas` does the same evaluation to ensure the iframe is "ready".

However, once we switch to Code View and then _back_ to Visual Mode the iframe has to "reload". Therefore using `editor.canvas` again without waiting results in the bug because it isn't "ready".

Reply:
Ah, that makes sense then - we need to wait for the frame explicitly until it remounts. And you're correct - it should not be needed once my PR lands!

Reply:
https://github.com/WordPress/gutenberg/pull/54911 just landed, so it should be safe to remove after rebase!

Reply:
@WunderBart I rebased and tests started failing. I think because I was using `editor.canvas` to query for elements in Code View mode. That mode doesn't seem to render an iframe so I guess that was the issue.

Updating all queries to use `page.getByRole` rather than `editor.canvas.getByRole` fixed the tests. 

See https://github.com/WordPress/gutenberg/pull/54718/commits/2260c27f57804dff09b3e48a2de0305596af2c43

Reply:
Nice, so the part in question wasn't actually needed in the end - it was the editor parent frame ambiguity that was at fault. Happy that it cleared out! 🙌